src/threadwrappers.cpp

```diff
@@ -37,13 +37,49 @@ using namespace dmtcp;
 
 extern __thread Thread *curThread;
 
-// Invoked via pthread_create as start_routine
-// On return, it calls mtcp_threadiszombie()
-static void *
-pthread_start(void *arg)
+static bool
+prepareThreadCreate()
 {
-  Thread *thread = (Thread *) arg;
+  /* pthread_create() should acquire the thread-creation lock. Not doing so can
```

Comment:
Minor:  `pthread_create()` -> `pthread_create and clone/clone3`

Can the first type of deadlock below still happen, now that pthread_create and clone are indpendent?